### ISSUE TITLE
"gm compare: Unable to open file" happening once/twice on almost every build

### ISSUE THREAD

**Issue by stefanteixeira** [oblador/loki#197]
Hi, I have a test suite with Loki but it's being really difficult to get it to pass. On almost every build I get at least one `gm compare: Unable to open file` issue with some random image.

Unfortunately the retry ability doesn't work with that kind of issue, so it's really frustrating to have almost every build failing. Does someone know what can I do to fix that?

**Comment by stefanteixeira** [comment]
I see this on our Jenkins builds. We store reference images on S3, so the builds fetch from S3 and then run Loki.

**Comment by oblador** [comment]
My guess is that the fetch from S3 somehow fails or isn't fully completed. Do you parallelize the fetch with the execution by any chance?

**Comment by stefanteixeira** [comment]
No, it's sequential 😕 
We have around ~400 images being tested, and this happens intermittently once or twice on almost every build. I had to do a workaround on the Jenkins job to check if logs contain only this error (and not all the other possible ones), then I set the exit status to 0 and make the job pass.

I'm closing this since it's really nebulous why this happens, but I managed to make jobs stable again with this workaround, so this is fine.
